src/build.d

```diff
@@ -317,6 +318,14 @@ alias sysconfDirFile = makeDep!((builder, dep) => builder
     })
 );
 
+alias mkdirDep = makeDepWithArgs!((MethodInitializer!Dependency builder, Dependency dep, string dir) => builder
```

Comment:
`directory` seems a bit too ambiguous for me.  It's true that the other dependencies are named based on what they generate rather than the action they perform, so it's inconsistent.  But in the context of a build program, the function `directory` could also mean that I want to create a directory object that can be used to make files or pass to functions that operate on a directory.

Reply:
Maybe not an optimal name at the declaration site but quite nice in a dependency list (IMO):
```
deps([ directory("results") ]);
```
But it's just a matter of personal preference feel free to keep as is.

It should have a DDOC comment though.

Reply:
I just don't like how ambiguous `directory` is.  Would you prefere `directoryDep` over `mkdirDep`?

Reply:
Yes